From 9d54454899766d34c2f22ccf5a1f955f547c1461 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 23 Oct 2012 12:54:24 +0800
Subject: [PATCH 606/609] kgdboc, omap: rx polling hook for the omap serial
 driver

The RX polling hook allows the debugger to hook character input so as
to allow entry to the kernel debugger with a control-c as an example.

Signed-off-by: Zumeng Chen <zumeng.chen@@windriver.com>
---
 drivers/tty/serial/omap-serial.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/tty/serial/omap-serial.c b/drivers/tty/serial/omap-serial.c
index 5a0b21d..712fcd6 100644
--- a/drivers/tty/serial/omap-serial.c
+++ b/drivers/tty/serial/omap-serial.c
@@ -237,6 +237,10 @@ static inline void receive_chars(struct uart_omap_port *up,
 				flag = TTY_FRAME;
 		}
 
+#ifdef CONFIG_CONSOLE_POLL
+		if (up->port.poll_rx_cb && up->port.poll_rx_cb(ch))
+			goto ignore_char;
+#endif
 		if (uart_handle_sysrq_char(&up->port, ch))
 			goto ignore_char;
 		uart_insert_char(&up->port, lsr, UART_LSR_OE, ch, flag);
-- 
1.7.5.4

